### PR TITLE
Feature/redux hamburger state

### PR DESCRIPTION
**Pull request by eouin**
use redux for hamburger state.
will add all search state later on.

### CODE REVIEW

File: src/components/Hamburger/Button/HamburgerButton.js [hunk 1]
```diff
@@ -1,14 +1,48 @@
...
   constructor(props) {
     super(props);
+
+    this.mediaQueryListener = null;
+    this.onMatchMQ = mediaQueryListener => this._onMatchMQ(mediaQueryListener);
+    this.toggleBodyStyle = toggle => this._toggleBodyStyle(toggle);
+  }
+
+  componentDidMount = () => {
+    if (!window.matchMedia) return;
```
Comment (ckoutsiaris) [inline src/components/Hamburger/Button/HamburgerButton.js]: Even IE10 supports window.matchMedia, why we need this?

Comment (eouin) [reply src/components/Hamburger/Button/HamburgerButton.js]: Done

**Comment by eouin** [comment]
@ckoutsiaris That's exactly what redux does. It removes all complexity in state management, removing the need to pass state thru all the components hierarchy. At the end of the day Redux is exactly what you saying, it's just a lib to maintain a global object.

**Comment by ckoutsiaris** [comment]
@eouin Yes I know but it also does a lot more that we probably don't need. Anyway, I am fine with getting experience in a popular library but I still think we don't need it.

**Pull request merged**